refactor: add day05 haha solution

haha861924 opened the pull request.

kayac-chang committed:
feat: 🎸 add day02 kirby

haha861924 committed:
feat: 🎸 initialize

haha861924 committed:
style: 💄 format

haha861924 committed:
feat: 🎸 add day04 haha

haha861924 committed:
fix: 🐛 no.7 sort

haha861924 committed:
feat: 🎸 initialize

haha861924 committed:
feat: 🎸 add click card transition

lee926427 commented on day05/haha/utils.js:
```diff
@@ -0,0 +1,22 @@
+// 展開收合 card ( toggle 自動切換)
+function clickToggle() {
+  this.classList.toggle("open");
+}
```
~~假設當使用者點開一個panel之後，再跑去點另一個panel會變成前一個點開的panel不會收回去唷~~
以上當我沒說過 剛剛跑去測一下原本就是這樣子XD

alphatero committed:
feat: 🎸  day 2 alpha's solution (#41)

* feat: 🎸 day one beginner

* feat: 🎸 add readme.md and build html first time

* feat: 🎸 add header component

* fix: 🐛 add dist

* fix: 🐛 sample assets link

* fix: 🐛 assets link fixed

* fix: 🐛 readme content

* feat: 🎸 first finsih before refactor

* refactor: 💡 day2 and add readme

* refactor: 💡 refresh header.js

* fix: 🐛 get hours deg error

* fix: 🐛 remove day1

* fix: 🐛 clock caculate function

* fix: 🐛 some issue

* fix: 🐛 build dist

haha861924 committed:
style: 💄 remove other days files

haha861924 committed:
feat: 🎸 add day05 readme

elzuoc committed:
try link markdown

elzuoc committed:
style: use markdown instead of table

haha861924 committed:
feat: add day02 haha solution (#10)

* initialize

* feat: get key code and remove class

* style: adjust structure

* style: format

* style: 💄 adjust structure

* feat: 🎸 initialize

* feat: 🎸 add clock

* fix: 🐛 hour hand angle

* style: 💄 format

* refactor: 💡 day02 answer

* Create README.md

* style: 💄 remove ohther days files

Co-authored-by: haha861924 <qq861924@gmail.com>

wen-hsiu-hsu committed:
add day 2 solution - Kevin (#43)

* feat: 🎸 day 2 first commit

* feat: 🎸 add day 2 solution

* docs: ✏️ update kevin day 2 readme

* fix: 🐛 setInterval 觸發時間修改

* 我就亂推，day01 多東西了啦

* 一些我要保留的修改

* 於是我就直接刪掉 day01 多的東西

* Revert "於是我就直接刪掉 day01 多的東西"

This reverts commit 39147c296b1988251879bcb42c32f6619008f75d.

* Revert "一些我要保留的修改"

This reverts commit f496e4e32434c3f7f8b44744ebdf07c5f6b3b09f.

* Revert "我就亂推，day01 多東西了啦"

This reverts commit 6d98fec273129d4cb023e8c20a3fdbf43c9159a4.

lee926427 committed:
feat: 🎸 add day02 Recoil (#51)

wen-hsiu-hsu committed:
docs: ✏️ add Day 2 README.md

lee926427 committed:
refactor: 💡  day02 Recoil (#73)

* refactor: semantic code

* docs: refactor README.md

ericacadu committed:
Feature/erica day02 (#68)

* feat: add day02 Erica solution

* update: add readme & script comment

* refactor: Erica day02

wen-hsiu-hsu committed:
refactor: 💡 Kevin day 2 (#67)

kayac-chang committed:
Feat/day02 kim (#63)

* feat: ✨[Pedding]

* feat: 🎸 [Pedding] Clock can move by second

Clock can move but time need fix.

BREAKING CHANGE: 🧨 no

* Create js

* feat: 🎸 add day02 kim

add day02 kim solution.

* Delete js

* remove unuse folder

* fix: 🐛 Feat/day02 kim

Remove console,  add HTML file, fix README.

Co-authored-by: Kimberly <yantingfen@yantingfendeMacBook-Air.local>
Co-authored-by: KimberlyYen <87320011+KimberlyYen@users.noreply.github.com>

wen-hsiu-hsu committed:
docs: ✏️ update day 2 readme - add few deployed links

HuskyHsu committed:
feat: 🎸 add day02 husky answer (#83)

* feat: 🎸 add day02 husky answer

* feat: 🎸 add transition timing function

ashramwen committed:
feat: 🎸 Add day02 cola solution (#87)

wen-hsiu-hsu committed:
docs: ✏️ update day 2 readme | add husky & cola page link

haha861924 committed:
Merge: day01 haha: adjust structure   (#2)

* initialize

* feat: get key code and remove class

* style: adjust structure

* style: format

* style: 💄 adjust structure

* refactor: 💡 replace keyCode with key value to ascii code

* Create README.md

* feat: 🎸 background img use sample file

* fix: 🐛 restart audio play time

* refactor: 💡 click transitionend remove checked class

Co-authored-by: haha861924 <qq861924@gmail.com>

alphatero committed:
refactor: 💡 renew branch (#85)

Merge: alpha's day01

HuskyHsu committed:
fix: 59 to 0 css transition bug (day02) (#91)

* feat: 🎸 add day02 husky answer

* feat: 🎸 add transition timing function

* fix: 🐛 59 to 0 css transition bug

* fix: 🐛 object literal extension

lee926427 committed:
feat: 🎸 add day01 Recoil (#50)

* feat: 🎸 add day01 Recoil

* docs: ✏️ add README.md

* fix: 🐛 change detection keyboard method

ezcomenezgo committed:
add day02 plusfive solution (#80)

Co-authored-by: Claire Li <claire.li@singularinfinity.com.tw>

beareatapple committed:
add Day02 White's homework (#82)

* add Day02 White's homework

I'm ready!

* fixed White's Day02

Thank for Kirby!

* fixed White's Day02 readme, delete unused function

I'm good rabbit.

* update White's Day02

忘記把自訂的pad函式刪掉，補刪除

wen-hsiu-hsu committed:
docs: ✏️ update day 2 readme | add white & plusfive

KimberlyYen committed:
add day01 kim (#31)

* feat: 🎸 [Pedding]first push

I will keep coding the project, not finished.tks

* feat: 🎸 add day01 kim

first commit

BREAKING CHANGE: 🧨 no

* feat: 🎸 add day01 kim

first commit

BREAKING CHANGE: 🧨 no

* fix: 🐛 README fixed

Writing my day01 solution at README.

* fix: 🐛 fixed day01 kim

- Remove IIFE

* fix: 🐛 fix sound files import

fix files import from sample.

Co-authored-by: Kimberly <yantingfen@yantingfendeMacBook-Air.local>

elzuoc committed:
day01-elzuoc-script.js: adjust variable name & function parameter (#56)

* day01-elzuoc-script.js: adjust variable name & function parameter

* feat: day01-use key instead keyCode

kayac-chang committed:
feat: 🎸 add day01 kirby solution (#7)

Merge: kirby's day01

wen-hsiu-hsu committed:
Day 1 - Kevin solution! (#24)

* feat: 🎸 day 1 kevin solution

* docs: ✏️ update day 1 readme

* refactor: 💡 改用 classList.toggle

* fix: 🐛 移除 event.keyCode 改用 event.key

* fix: 🐛 移除 IIFE

ericacadu committed:
Feature/erica day01 (#46)

* add day01 Erica answer

* update day01 Erica answer & add readme

* add day01 Erica answer

* update day01 Erica answer & add readme

* fix: revise day01 solution, add try...catch

* update day01 Erica answer & add readme

* update day01 erica file

* fix: git conflict

HuskyHsu committed:
[feature] add day01 husky (#55)

* feat: 🎸 day01 init

* feat: 🎸 update day01 answer

* feat: 🎸 day01 add addEventListener event

* feat: 🎸 day01 done

* feat: 🎸 day01 init

* feat: 🎸 update day01 answer

* feat: 🎸 day01 add addEventListener event

* feat: 🎸 day01 done

* fix: 🐛 fix pull request comment of react suggest

zoeGuava committed:
Feature/day01 pencil's solution (#72)

* feat: day01 initial

* update: day01 done

beareatapple committed:
add Day01 White (#74)

Finished Day01 White's solution

ezcomenezgo committed:
Feature: add day01 plusfive v2 solution (#89)

* feat. add day 1 JS Drum Kit plusfive's solution

* fix. 修改 keyCode 改用 key + charCodeAt()

* 修改 README

* 改用 Map 的方式存取及加上監聽 keyup 事件

* 修改 README

Co-authored-by: Claire Li <claire.li@singularinfinity.com.tw>

ericacadu committed:
Feature/erica day03 (#79)

* feature: add day03 Erica solution

* fix: change input event

wen-hsiu-hsu committed:
Feat: kevin day3 solution (#81)

* feat: 🎸 first commit on day3

* feat: 🎸 day3 main feature done

* docs: ✏️ add kevin day3 readme

* feat: 🎸 select dom by class instead of tag

* feat: 🎸 use Nullish coalescing operator

lee926427 committed:
feat: 🎸 add day03 Recoil (#90)

* feat: 🎸 add day03 Recoil

抱歉 push時被要求pull不小心合併到分支上，在此奉上十份奶捲

* refactor: 💡 add initStyleEffect func

HuskyHsu committed:
fix: 🐛 add day03 husky answer (#93)

haha861924 committed:
feat: add day03 haha solution (#12)

* initialize

* feat: get key code and remove class

* style: adjust structure

* style: format

* style: 💄 adjust structure

* feat: 🎸 initialize

* feat: 🎸 add img css dynamic style

* style: 💄 remove other days files

* feat: 🎸 add loss css style

* refactor: 💡 day03 answer

* Create README.md

Co-authored-by: haha861924 <qq861924@gmail.com>

zoeGuava committed:
Feature/day03 pencil (#94)

* feat: initial file

* update: add css variables and README

* update: prettier the index.html

* update: pencil's day03 css variable done...

* update: add README

kayac-chang committed:
feat: 🎸 add day03 kirby (#9)

ezcomenezgo committed:
add Day03 plusfive solution (#88)

Co-authored-by: Claire Li <claire.li@singularinfinity.com.tw>

zoeGuava committed:
Update README.md

KimberlyYen committed:
feat: 🎸 add day03 kim (#78)

* feat: 🎸 add day03 kim

add day03 kim

* feat: 🎸 add day02 husky answer (#83)

* feat: 🎸 add day02 husky answer

* feat: 🎸 add transition timing function

* feat: 🎸 Add day02 cola solution (#87)

* docs: ✏️ update day 2 readme | add husky & cola page link

* Merge: day01 haha: adjust structure   (#2)

* initialize

* feat: get key code and remove class

* style: adjust structure

* style: format

* style: 💄 adjust structure

* refactor: 💡 replace keyCode with key value to ascii code

* Create README.md

* feat: 🎸 background img use sample file

* fix: 🐛 restart audio play time

* refactor: 💡 click transitionend remove checked class

Co-authored-by: haha861924 <qq861924@gmail.com>

* refactor: 💡 renew branch (#85)

Merge: alpha's day01

* fix: 59 to 0 css transition bug (day02) (#91)

* feat: 🎸 add day02 husky answer

* feat: 🎸 add transition timing function

* fix: 🐛 59 to 0 css transition bug

* fix: 🐛 object literal extension

* feat: 🎸 add day01 Recoil (#50)

* feat: 🎸 add day01 Recoil

* docs: ✏️ add README.md

* fix: 🐛 change detection keyboard method

* feat: 🎸 add day03 kim

add day03 kim

* [Pedding]

* fix: 🐛 fix branch

Delete Day04 at this branch

* fix: 🐛 remove console.log, fix path of HTML

HTML path add "." become ./style.css

* fix: 🐛 README fixed

* remove day04

Co-authored-by: Kimberly <yantingfen@yantingfendeMacBook-Air.local>
Co-authored-by: ShihChi, Hsu <schi.hsu@gmail.com>
Co-authored-by: Augustine <ashramwen@users.noreply.github.com>
Co-authored-by: kevin_hws <kevinshu1995@gmail.com>
Co-authored-by: haha861924 <40877568+haha861924@users.noreply.github.com>
Co-authored-by: haha861924 <qq861924@gmail.com>
Co-authored-by: Alpha <46735230+alphatero@users.noreply.github.com>
Co-authored-by: lee926427 <40841221+lee926427@users.noreply.github.com>

lee926427 commented on day05/haha/utils.js:
```diff
@@ -0,0 +1,22 @@
+
+// 展開收合 card ( toggle 自動切換)
+function clickToggle() {
+  this.classList.toggle("open");
```
avoid use `this`, another way use for  arguments
```js
function clickToggle({target}){
  target.classList.toggle("open")
}
```

haha861924 replied:
this will not work

haha861924 replied:
In this way, this event listener will trigger other element tag  
![image](https://user-images.githubusercontent.com/40877568/149552620-90fe8d26-ee61-4719-948e-0844f5c7510b.png)

haha861924 replied:
I suggest here neet to another ways to realize it
I will refactor it , thanks your good idea

lee926427 replied:
Here recommend two way for you.
JS: [closest](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) 
CSS: [pointer-events](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events)

haha861924 replied:
I already use another way to refact it

haha861924 replied:
I don't know why you tag [pointer-events](https://developer.mozilla.org/en-US/docs/Web/CSS/pointer-events)  for me
This way can help me what?

haha861924 replied:
[closest](https://developer.mozilla.org/en-US/docs/Web/API/Element/closest) is a good another way to find node.
It can help me just one step replace closest with confirm nodeName to be current node or its parent node.
I learned it !!!! thx!

lee926427 replied:
use `pointer-events: none;` the element trigger will ignored.

haha861924 replied:
ok I got it . In my new way is no longer trigger and I avoid use this, and I learned a new idea, thxs

alphatero committed:
feat: 🎸 add alpha's day3 solution (#99)

* feat: 🎸 add alpha day03 solution

* feat: 🎸 finished day3 solution

* feat: 🎸 add dist

* docs: ✏️ add day3 readme

haha861924 committed:
fix day03 immediately variety (#101)

* initialize

* feat: get key code and remove class

* style: adjust structure

* style: format

* style: 💄 adjust structure

* feat: 🎸 initialize

* feat: 🎸 add img css dynamic style

* style: 💄 remove other days files

* feat: 🎸 add loss css style

* refactor: 💡 day03 answer

* Create README.md

* fix: 🐛 immediate input Variety

Co-authored-by: haha861924 <qq861924@gmail.com>
Co-authored-by: zoeGuava <49506360+zoeGuava@users.noreply.github.com>

lee926427 committed:
refactor: 💡 day03 Recoil (#104)

lee926427 reviewed (changes requested):
Good job

zoeGuava committed:
Update README.md

KimberlyYen committed:
Revert "remove day04" (#112)

This reverts commit 11f292290c575686840503ed85e399434d0622c7.

alphatero committed:
docs: ✏️ add speaker (#113)

ashramwen committed:
chore: 🤖 Add lint (#121)

ashramwen committed:
chore: Format js files (#122)

* chore: 🤖 Update prettier settings

* chore: 🤖 prettier

ashramwen committed:
chore: 🤖 Updage package (#124)

ashramwen committed:
Create auto-author-assign.yml

kayac-chang committed:
fix: 🐛 fix init and missing element (#107)

haha861924 committed:
docs: ✏️ update README

haha861924 committed:
refactor: 💡 change select element

haha861924 committed:
refactor: 💡 click panel

haha861924 committed:
refactor: 💡 panel transition end handle event

haha861924 committed:
style: 💄 format .html

haha861924 committed:
test: 💍 rebase then commit

haha861924 committed:
style: 💄 the third format

haha861924 commented:
![image](https://user-images.githubusercontent.com/40877568/149616047-1d746e3d-82ac-41f7-8ee2-b91e76ff061e.png)
